### ISSUE TITLE
munmap_chunk(): invalid pointer on api_.start() in mono_camera.cpp

### ISSUE THREAD

**Issue by rcywongaa** [astuff/avt_vimba_camera#5]
The same thing happens when using the newer vimba API, where the same error is thrown on
```
m_system.Startup();
```
Where `m_system` is initialized as follows
```
    AVT::VmbAPI::VimbaSystem &       m_system(AVT::VmbAPI::VimbaSystem::GetInstance()) ;                   // A reference to our Vimba singleton
    m_system.Startup();
```

Issue cross posted in original repo as well: https://github.com/srv/avt_vimba_camera/issues/33

**Comment by icolwell-as** [comment]
Thanks for reporting this, unfortunately we don't have any USB hardware to test/reproduce with. We only use the Ethernet mako cameras.

**Comment by StephanEll** [comment]
I have the same problem. Did anyone find a solution?

**Comment by rcywongaa** [comment]
@StephanEll I have contacted Allied Vision on this and have verified that it is an issue with their USB Transport Layer which they'll have to fix on their side.  It appears they are aware of issues with using their cameras with ROS.  Unfortunately, it doesn't seem high on their priority list and I haven't heard from them since.

**Comment by mrkschnk** [comment]
@StephanEll , @rcywongaa Allied Vision is very interested in enabling the ROS community to use Vimba with ROS. But since we are not the experts in ROS, we need you to find the problem.
There may be a conflict with external libraries when using the USB interface with ROS. To further narrow this down, it would be great if you could communicate your used system configuration (like Host system / architecture, version from OS, Vimba, ROS, libboost, OpenCV) to the Allied vision support.

**Comment by rcywongaa** [comment]
@mrkschnk That's good to know.

I've spoken with one of Allied Vision's application engineer (Jan Lemke).  He was able to reproduce the issue on his side and has confirmed that it was an issue with the transport layer provided by the vimba library.  I believe he has reported it to the relevant product manager already.  He also stated that it was a known issue that there are compatibility issues with ROS and Vimba.
Below is a sample ROS project that exhibits the problem (which I've also sent to Jan).
https://github.com/rcywongaa/vimba_invalid_pointer

**Comment by mrkschnk** [comment]
@StephanEll , @rcywongaa Allied Vision released a new Vimba version 4.1 to address your experienced problem. Happy to get your feedback on this: https://www.alliedvision.com/en/products/software.html

**Comment by icolwell-as** [comment]
Closing since the issue appears to be resolved.
